docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+This creates a copy of the code under your GitHub user account.
+
+2. Clone your fork to your local disk, and add the base repository as a remote:
+
+	```bash
+	git clone https://github.com/[your Github handle]/transformers.git
+	cd transformers
+	git remote add upstream https://github.com/huggingface/transformers.git
+	```
+ 
+3. Set up a development environment, for instance by running the following command:
+
+	```bash
+	conda create -n env python=3.7 --y
+	conda activate env
+	pip install -e ".[dev]"
+	```
+  
+  and return to the parent directory
+  
+  ```bash
+  cd ..
+  ```
+
+4. We recommend to add the PyTorch version of brand_new_bert to Transformers. To install PyTorch,
+  please follow the instructions on https://pytorch.org/. 
+  
+  **Note:** You don't need to have CUDA install. It is sufficient to just be working on CPU.
+
+5. To port brand_new_bert, you will also need access to its original repository:
+  
+  ```bash
+  git clone https://github.com/brand_new_bert_org/brand_new_bert.git 
+  cd brand_new_bert
+  pip install -e .
+  ```
+
+Now you have set up a development environment to port brand_new_bert to 🤗Transformers.
+  
+### Run a pretrained checkpoint using the original repository
+
+At first, you will work on the original repository. Often, the original implementation is very
+ "researchy" meaning that documentation might be lacking and the code can be difficult to understand.
+But this should be exactly your motivation to reimplement brand_new_bert. At Hugging Face, one of our main goals is to
+*make people stand on the shoulders of giants* which translates here very well into taking a working 
+model and rewriting it to make it as **accessible, user-friendly, and beautiful** as possible.
+This is the #1 motivation to re-implement models into 🤗Transformers - trying to maximize access 
+to a complex new NLP technology for **everybody**.
+	
+You should start thereby by diving into the original repository.
+
+#### Get familiar with the original repository.
+
+Successfully running the official pre-trained model in the original repository is often 
+**the most difficult** step. From our experience, it is very important to spend some time to get familiar with the original codebase. You should find out
+
+- Where to find the pre-trained weights
+- How to load the pre-trained weights into its corresponding model
+- Trace one forward pass so that you know which classes and functions are required for a simple forward pass
+  . Usually, you only have to reimplement those functions.
+- Be able to locate the important components of the model: Where is the model class? Are there submodel 
+  classes, *e.g.* EncoderModel, DecoderModel? Where is the self-attention layer? 
+  Are there multiple different attention layers, *e.g.* *self-attention*, *cross-attention*...?
+- How can you debug the model in the original environment of the repo? Do you have to set `print` statements
+  or can you work with an interactive debugger like `ipdb`?
```

Comment:
```suggestion
- Where to find the pre-trained weights?
- How to load the pre-trained weights into its corresponding model?
- Trace one forward pass so that you know which classes and functions are required for a simple forward pass
  . Usually, you only have to reimplement those functions.
- Be able to locate the important components of the model: Where is the model class? Are there submodel 
  classes, *e.g.* EncoderModel, DecoderModel? Where is the self-attention layer? 
  Are there multiple different attention layers, *e.g.* *self-attention*, *cross-attention*...?
- How can you debug the model in the original environment of the repo? Do you have to set `print` statements
  or can you work with an interactive debugger like `ipdb`?
```